instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/logs/GdprLogsE2ETests.kt

```diff
@@ -0,0 +1,160 @@
+        }
+        logger.sendRandomLog(
+            testMethodName,
+            forge
+        )
+    }
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun logs_config_consent_not_granted() {
+        val testMethodName = "logs_config_consent_not_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.NOT_GRANTED
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+
+    /**
+     * apiMethodSignature: Datadog#fun setTrackingConsent(com.datadog.android.privacy.TrackingConsent)
+     */
+    @Test
+    fun logs_config_consent_pending_to_granted() {
+        val testMethodName = "logs_config_consent_pending_to_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.PENDING
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+        measureSetTrackingConsent {
+            Datadog.setTrackingConsent(TrackingConsent.GRANTED)
+        }
+    }
+
+    /**
+     * apiMethodSignature: Datadog#fun setTrackingConsent(com.datadog.android.privacy.TrackingConsent)
+     */
+    @Test
+    fun logs_config_consent_pending_to_not_granted() {
+        val testMethodName = "logs_config_consent_pending_to_not_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.PENDING
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+        measureSetTrackingConsent {
+            Datadog.setTrackingConsent(TrackingConsent.NOT_GRANTED)
+        }
+    }
+
+    /**
+     * apiMethodSignature: Datadog#fun setTrackingConsent(com.datadog.android.privacy.TrackingConsent)
+     */
+    @Test
+    fun logs_config_consent_granted_to_not_granted() {
+        val testMethodName = "logs_config_consent_granted_to_not_granted"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                consent = TrackingConsent.GRANTED
+            )
+        }
+        measureLoggerInitialize {
+            logger = initializeLogger()
+        }
+        logger.sendRandomLog(testMethodName, forge)
+        measureSetTrackingConsent {
+            Datadog.setTrackingConsent(TrackingConsent.NOT_GRANTED)
+        }
+        logger.sendRandomLog(testMethodName, forge)
+    }
+}
```

Comment:
I missed them intentionally. Having in mind that these particular transitions will not produce any event I thought it would be wiser to omit those and to avoid adding any unnecessary monitors. I think we should adopt a best practice and only add a monitor when it really makes sense. WDYT @xgouchet @0xnm ?